src/main/java/com/example/compound/controller/Controller.java

```diff
@@ -459,62 +426,89 @@ private void updateGroup(InOut inOut) {
                 inOut.sendOutput("Please enter a valid group name.\n");
                 break;
             }
-            Group g = groupManager.getGroupByName(groupName);
-            assert g != null;
+            String GUID = groupManager.getGUIDFromName(groupName);
+            currentGroupManager.setCurrentGroup(GUID);
             int inputG = inOut.getActionView(groupActions);
-            back = manageGroup(inOut, back, g, inputG);
+            back = manageGroup(inOut, false, currentGroupManager, inputG);
         }
 
     }
 
+    /**
+     * A helper method that enables current user to add or remove people to a group.
+     * @param inOut The interface for the view
+     * @param addOrRemove Add or Remove
+     * @return A list of strings of people to be added or removed
+     */
+    private List<String> addRemovePeople(InOut inOut, String addOrRemove) {
+        List<String> people = new ArrayList<>();
+        boolean addRemovePeople = Boolean.TRUE;
+        do {
+            inOut.sendOutput("Do you want to " + addOrRemove + " more people in your "
+                    + "group" + "? (y/n)");
+            String input2 = inOut.getInput();
+            switch (input2) {
+                case "y" -> {
+                    inOut.sendOutput("Enter user email:");
+                    people.add(inOut.getInput());
+                }
+                case "n" -> addRemovePeople = Boolean.FALSE;
+            }
+        } while (addRemovePeople);
+        return people;
+    }
+
+
     /**
      * A helper method extracted from the updateGroup method that involves all the optional actions on groups.
      * @param inOut the user interface object
      * @param back the while-loop "indicator"
-     * @param g the group that the user selected
+     * @param CGM the group that the user selected
      * @param inputG the option that the user chose
      * @return the while-loop "indicator"
      */
-    private boolean manageGroup(InOut inOut, boolean back, Group g, int inputG) {
+    public boolean manageGroup(InOut inOut, boolean back, CurrentGroupManager CGM, int inputG) {
         switch (inputG){
             case 1 -> {
                 inOut.sendOutput("Please enter the new name.");
                 String newName = inOut.getInput();
-                GroupManager.setGroupName(g, newName);
+                GroupManager.setGroupName(CGM.getCurrentGroup(), newName);
```

Comment:
Instead of taking in a Group object as an argument, the setGroupName method (and other GroupManager methods in other cases) could take in either the CurrentGroupManager or the GUID of the current group, like in BudgetController. Passing in just the GUID would mean only simple data structures would be passed across the boundary between the Application Business Rules and the Interface Adapters, but it would also mean that the GroupManager would have to repeatedly get the Group object from the repository using the GUID (like BudgetManager).

Reply:
Okay. But what did you mean by repeatedly getting the Group object from the repository using the GUID?

Reply:
BudgetManager's methods all call findByGUID, findByBUID, getBudgets, or getItems in RepositoryGateway. An alternative might be storing CurrentBudgetManager and CurrentGroupManager instances in BudgetManager, passing them in through BudgetManager's constructor, to make some of those calls unnecessary. Similarly, a CurrentGroupManager instance could be stored in GroupManager instead of passing the current group or its GUID into the setGroupName method.

Reply:
I've modified that. Please take a look of my new commits.